src/pages/Stream/Views/Manage/Info.tsx

```diff
@@ -13,35 +17,221 @@ const Header = () => {
+	const partitionFields: string[] = updateableCustomPartition.split(',');
+	const [value, setValue] = useState<string[]>(partitionFields);
+	const [updating, setUpdating] = useState<boolean>(false);
+	const { updateLogStreamMutation } = useLogStream();
+	const { getStreamInfoRefetch } = useGetStreamInfo(props.currentStream);
+
+	const onChangeValue = useCallback(
+		(value: string[]) => {
+			setValue(value);
+		},
+		[setValue],
+	);
+
+	const updateLogStreamSuccess = useCallback(() => {
+		props.onClose();
+		setUpdating(false);
+		getStreamInfoRefetch();
+	}, [getStreamInfoRefetch]);
+
+	const updateLogStream = useCallback(
+		(updatedValue: string) => {
+			updateLogStreamMutation({
+				streamName: props.currentStream,
+				customPartition: updatedValue,
+				onSuccess: () => updateLogStreamSuccess(),
+				onError: () => setUpdating(false),
+			});
+		},
+		[props.currentStream, updateLogStreamMutation],
+	);
+
+	const onClick = useCallback(() => {
+		if (value === undefined) return;
+
+		const valuesFlattened = value.join(',');
+		setUpdating(true);
+		updateLogStream(valuesFlattened);
+	}, [value, updateLogStream]);
+
+	return (
+		<Stack gap={0} style={{ justifyContent: 'space-between' }}>
+			<Stack>
+				<Text className={classes.fieldDescription}>Custom Partition Field</Text>
+			</Stack>
+
+			<Stack style={{ flexDirection: 'row', alignItems: 'center' }}>
+				<TagsInput
+					w={'30rem'}
+					placeholder="Add upto 3 columns"
+					data={partitionFields}
+					onChange={(value) => onChangeValue(value)}
+					maxTags={3}
+					value={value}
+				/>
+				<UpdateFieldButtons onClick={onClick} onClose={props.onClose} updating={updating} />
+			</Stack>
 		</Stack>
 	);
 };
 
-const InfoData = (props: {isLoading: boolean}) => {
+const InfoItem = (props: { title: string; value: string; fullWidth?: boolean; editable?: boolean }) => {
+	const [showEditField, setShowEditField] = useState<boolean>(false);
+	const [currentStream] = useAppStore((store) => store.currentStream);
+	const closeEditField = useCallback(() => {
+		setShowEditField(false);
+	}, []);
+
+	return (
+		<>
+			{!showEditField ? (
+				<Stack w={props.fullWidth ? '100%' : '33%'} gap={14}>
+					<Group gap={6}>
+						<Text
+							className={classes.fieldDescription}
+							style={{ textOverflow: 'ellipsis', whiteSpace: 'nowrap', overflow: 'hidden' }}>
+							{props.title}
+						</Text>
+						{props.editable ? (
+							<Tooltip label="Edit" withArrow position="top">
+								<IconEdit
+									className={classes.infoEditBtn}
+									stroke={1.6}
+									size={12}
+									onClick={() => setShowEditField(true)}
+								/>
+							</Tooltip>
+						) : null}
+					</Group>
+					<Text
+						className={classes.fieldTitle}
+						style={{ textOverflow: 'ellipsis', whiteSpace: 'nowrap', overflow: 'hidden', fontWeight: 400 }}>
+						{props.value}
+					</Text>
+				</Stack>
+			) : props.title === 'Custom Partition Field' ? (
```

Comment:
Checking for this one condition is not a good call here. 
What if I want to add edit functionality to another field?